L1Trigger/TrackFindingTracklet/plugins/ProducerKFout.cc

```diff
@@ -109,19 +161,139 @@ namespace trackFindingTracklet {
+                SortedPartialTracks[iLink].push_back(PartialTrack2);
+                SortedPartialTracks[iLink].push_back(PartialTrack3);
+                OutputStreamsTracks[iLink].emplace_back(track);
+              }
+              else{
+                SortedPartialTracks[iLink+1].push_back(PartialTrack1);
+                SortedPartialTracks[iLink+1].push_back(PartialTrack2);
+                SortedPartialTracks[iLink+1].push_back(PartialTrack3);
+                OutputStreamsTracks[iLink+1].emplace_back(track);
+              }
+            }
+          else{
+            if (InTrack.sectorEta() < (int)(setup_->numSectorsEta()/2)){
+              SortedPartialTracks[iLink-1].push_back(PartialTrack1);
+              SortedPartialTracks[iLink-1].push_back(PartialTrack2);
+              SortedPartialTracks[iLink-1].push_back(PartialTrack3);
+              OutputStreamsTracks[iLink-1].emplace_back(track);
+            }
+            else{
+              SortedPartialTracks[iLink].push_back(PartialTrack1);
+              SortedPartialTracks[iLink].push_back(PartialTrack2);
+              SortedPartialTracks[iLink].push_back(PartialTrack3);
+              OutputStreamsTracks[iLink].emplace_back(track);
+            }
+ 
+          }
+        } // Tracks
+      } // Links
+
+      // Fill products and match up tracks
+      TTBV NullBitTrack(0,32,false);
+      for (int iLink = 0; iLink < (int)OutputStreamsTracks.size(); iLink++ ){
+        // Iterate through partial tracks
+        int numLinkTracks = (int)OutputStreamsTracks[iLink].size();
+        if (numLinkTracks > 0){
+          if ((numLinkTracks % 2 != 0)) { //If there is an odd number of tracks 
+            SortedPartialTracks[iLink].push_back(NullBitTrack);  //Pad out final set of bits
+            OutputStreamsTracks[iLink].emplace_back(OutputStreamsTracks[iLink][numLinkTracks]); //Pad out with final repeated track
+            numLinkTracks++;
+            } //If there is an odd number of tracks 
+          for (int iTrack = 0; iTrack < (int)(SortedPartialTracks[iLink].size()); iTrack++ ){  
+            if (iTrack % 2 == 1){
+              TTTrackRef TrackRef;
+              for (auto &it : ttTrackRefMap) {   //Iterate throguh ttTrackRefMap to find TTTrackRef Key by a TTTrack Value
+                if(it.second == OutputStreamsTracks[iLink][(int)(iTrack-1)/3].first) { 
+                  TrackRef = it.first;
+                } 
+              }
+              if ((int)iTrack/3 <= maxTracksPerEvent_){
+                accepted[iLink].emplace_back(std::make_pair(TrackRef,(SortedPartialTracks[iLink][iTrack].slice(32) + SortedPartialTracks[iLink][iTrack-1].slice(32)).bs()));
+              }
+              else{
+                lost[iLink].emplace_back(std::make_pair(TrackRef,(SortedPartialTracks[iLink][iTrack].slice(32) + SortedPartialTracks[iLink][iTrack-1].slice(32)).bs()));
+              }
+            }
+          }
+        }
+      }
```

Comment:
I personally prefer less indented code. In case where you have an
```
if (A) {
  bla;
}
```
without else branch could you consider
```
if (!A)
  continue;
bla;
```